src/kiwixapp.cpp

```diff
@@ -93,6 +100,19 @@ void KiwixApp::init()
             this->openZimFile(message);
         }
     });
+    connect(&m_watcher, &QFileSystemWatcher::directoryChanged, this, [=](QString monitorDir) {
+        m_library.loadMonitorDir(monitorDir);
+    });
```

Comment:
After rebasing and fixing-up this change appears earlier than the commit in which `m_watcher` is introduced. My personal protection against such issues after non-trivial history clean-up is to replay the history and make sure that each commit compiles (and doesn't break the unit tests).